source/common/http/http1/codec_impl.cc

```diff
@@ -244,28 +244,36 @@ http_parser_settings ConnectionImpl::settings_{
     nullptr  // on_chunk_complete
 };
 
-ConnectionImpl::ConnectionImpl(Network::Connection& connection, http_parser_type type)
-    : connection_(connection) {
-  http_parser_init(&parser_, type);
-  parser_.data = this;
+const ConnectionImpl::ToLowerTable ConnectionImpl::to_lower_table_;
+
+ConnectionImpl::ToLowerTable::ToLowerTable() {
+  for (size_t c = 0; c < 256; c++) {
+    table_[c] = c;
+    if ((c >= 'A') && (c <= 'Z')) {
+      table_[c] |= 0x20;
+    }
+  }
 }
 
-static void toLowerCase(HeaderString& text) {
+void ConnectionImpl::ToLowerTable::toLowerCase(HeaderString& text) const {
   char* buffer = text.buffer();
   uint32_t size = text.size();
   for (size_t i = 0; i < size; i++) {
-    char c = buffer[i];
-    if ((c >= 'A') && (c <= 'Z')) {
-      buffer[i] |= 0x20;
-    }
+    buffer[i] = table_[static_cast<uint8_t>(buffer[i])];
```

Comment:
Been a while since I optimized at this level, but would it help to use `restrict` on `buffer` to indicate that it can't overlap with `table_` ?